Fix instructions and Makefiles for IPASIR-based SAT solvers

tautschnig opened the pull request:
The LIBSOLVER variable didn't actually work, because it was not used
consistently. Instead, using LIBS suffices. For example, to use RISS,
build as follows:

git clone https://github.com/conp-solutions/riss riss.git
cd riss.git
mkdir build
cd build
cmake ..
make riss-coprocessor-lib-static
cd ../..
make -C src IPASIR=../../riss.git/riss \
  LIBS="../../riss.git/build/lib/libriss-coprocessor.a -lpthread"
export RISSCONFIG=VERBOSE:BMC1
... run CBMC ...

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Fix instructions and Makefiles for IPASIR-based SAT solvers

The LIBSOLVER variable didn't actually work, because it was not used
consistently. Instead, using LIBS suffices. For example, to use RISS,
build as follows:

git clone https://github.com/conp-solutions/riss riss.git
cd riss.git
mkdir build
cd build
cmake ..
make riss-coprocessor-lib-static
cd ../..
make -C src IPASIR=../../riss.git/riss \
  LIBS="../../riss.git/build/lib/libriss-coprocessor.a -lpthread"
export RISSCONFIG=VERBOSE:BMC1
... run CBMC ...

tautschnig commented:
I could add the instructions included in the commit message as a new task in Travis, but I'm not sure this is desirable.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 9d23c0128293eb0da9f80805c904aabd35d80e02).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85208213

peterschrammel reviewed (approved):
Your example of how to use a different sat solver could be nicely put into a subsection here: https://github.com/diffblue/cbmc/blob/develop/doc/architectural/compilation-and-development.md (Maybe in a follow-up PR).

tautschnig committed:
Document the IPASIR build in a single place

As witnessed by the prior commit, errors in the instructions had to be
updated in multiple places. Also removing the Makefile-based build rules
for IPASIR as those would only work for outdated solvers.

tautschnig commented:
> Your example of how to use a different sat solver could be nicely put into a subsection here: [...]

@peterschrammel I have added a second commit to this PR - please take a look if that looks about right. It does include further cleanup.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 3f4a89621a9693316c302796a0fe8421cf87ca92).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85257255

peterschrammel reviewed (approved):
Thank you!
